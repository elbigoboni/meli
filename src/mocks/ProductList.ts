import Product from "../types/Product";

export const products: Product[] = [
  {
    title: "Produto 1",
    description: "lalalalalal",
    price: 2000,
    location: "Lugar do produto",
    image: {
      full: "//placeimg.com/680/680",
      thumb: "//placeimg.com/170/170",
    },
  },
  {
    title: "Produto 1",
    description: "lalalalalal",
    price: 2000,
    location: "Lugar do produto",
    image: {
      full: "//placeimg.com/680/680",
      thumb: "//placeimg.com/170/170",
    },
  },
  {
    title: "Produto 1",
    description: "lalalalalal",
    price: 2000,
    location: "Lugar do produto",
    image: {
      full: "//placeimg.com/680/680",
      thumb: "//placeimg.com/170/170",
    },
  },
  {
    title: "Produto 1",
    description: "lalalalalal",
    price: 2000,
    location: "Lugar do produto",
    image: {
      full: "//placeimg.com/680/680",
      thumb: "//placeimg.com/170/170",
    },
  },
];
