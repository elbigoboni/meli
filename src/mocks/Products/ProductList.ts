import ProductInteface from "../../types/Product";
import ProductDetailInteface from "../../types/ProductDetail";

export const ProductsListMock: ProductInteface[] = [
  {
    author: {
      name: "Author xyz",
    },
    item: {
      id: "ml1762712761",
      title: "macbook",
      categories: ["cat1", "cat2"],
      thumbnail: "string",
      price: {
        amount: 1200,
        currency: "ARS",
      },
      location: "Buenos Aires",
      condition: "new",
      free_shipping: true,
    },
  },
  {
    author: {
      name: "Author xyz2",
    },
    item: {
      id: "ml8718728712",
      title: "ipad",
      categories: ["cat1", "cat2"],
      thumbnail: "string",
      price: {
        amount: 500,
        currency: "ARS",
      },
      location: "San Francisco",
      condition: "new",
      free_shipping: true,
    },
  },
];

export const ProductDetailMock: ProductDetailInteface = {
  author: {
    name: "Author xyz",
  },
  description: "desc xyz",
  item: {
    id: "ml1762712761",
    title: "macbook",
    categories: ["cat1", "cat2"],
    thumbnail: "string",
    price: {
      amount: 1200,
      currency: "ARS",
    },
    location: "Buenos Aires",
    condition: "new",
    free_shipping: true,
  },
  sold_quantity: 12,
};
