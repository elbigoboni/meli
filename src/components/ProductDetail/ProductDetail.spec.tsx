import { render } from "@testing-library/react";
import React from "react";
import { ProductDetail } from ".";
import { products } from "../../mocks/ProductList";

describe("<ProductDetail />", () => {
  describe("Render", () => {
    test("should render correctly", () => {
      const { container } = render(<ProductDetail product={products[0]} />);
      expect(container).toMatchSnapshot();
    });
  });
});
