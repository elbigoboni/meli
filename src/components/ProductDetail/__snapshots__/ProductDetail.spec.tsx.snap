// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDetail /> Render should render correctly 1`] = `
<div>
  <div
    class="sc-gsTCUz cIVwHc"
  >
    <div
      class="sc-gKsewC iXHTOo"
    >
      <img
        class="sc-pFZIQ brYlDS"
        src="https://images-submarino.b2w.io/produtos/01/00/img/1803404/3/1803404367_1SZ.jpg"
      />
    </div>
    <div
      class="sc-iBPRYJ eHxXYq"
    >
      <h1
        class="sc-dlfnbm FLZNm"
      >
        iPad Pro
      </h1>
      <p
        class="sc-hKgILt RUpaM"
      >
        R$ 5.893
      </p>
      <button
        class="sc-bdfBwQ cMSeGt"
      >
        Comprar
      </button>
    </div>
    <div
      class="sc-fubCfw gOFgOY"
    >
      <h2
        class="sc-eCssSg cypuFD"
      >
        Descripción del producto
      </h2>
      <p
        class="sc-jSgupP esWSIU"
      >
        Seu próximo computador não é um computador. Ele é uma superfície quase mágica. Mais rápido que a maioria dos laptops PC. Com câmeras que transformam a realidade. E você pode tocar, usar com Apple Pencil, com teclado e até com trackpad. É o novo iPad Pro.
      </p>
    </div>
  </div>
</div>
`;
